### PR TITLE
Handle POST reqs and persist data changes

### PR DESCRIPTION
**Pull request by alshakerM**

### CODE REVIEW

File: mocking-scripts/add-tags-to-media.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        return photo;
+      });
+    post.post_has_tags = true;
+  }
+  post.post_has_tags = false;
+  return post;
+});
+
+
+writeFileSync('../server/posts-data.json', JSON.stringify(posts));
```
Comment (alshakero) [inline mocking-scripts/add-tags-to-media.js:32]: This is unrelated to this PR.

File: pages/api/messages.js [hunk 2]
```diff
@@ -9,45 +22,102 @@ const allowedFields = [
...
 
-        return filteredThread;
-      });
+    if (!fromUserId) {
+      return res
+        .status(400)
+        .send('Bad request. fromUserId needs to be provided');
+    }
+    if (typeof index !== 'number') {
+      return res.status(400).send('Bad request. index needs to be a number');
```
Comment (alshakero) [inline pages/api/messages.js:35]: This is causing an issue with `submitMessage` action, it doesn't have an index. Move the check to where index is used.

File: pages/api/messages.js [hunk 3]
```diff
@@ -9,45 +22,102 @@ const allowedFields = [
...
+          is_liked_by_user: false,
+          direction: 'sent',
+        });
+        syncMessageFile();
+        return res
+          .status(200)
+          .json({ ok: true, message: 'Message is now Added' });
+      }
+      default: {
+        return res.status(400).send(`Bad request. Bad action type: ${action}`);
```
Comment (alshakero) [inline pages/api/messages.js:77]: I just realized that sending the wrong action back to the user is a security threat. Let's remove this.
```suggestion
        return res.status(400).send(`Bad request. Bad action type.`);
```

File: pages/api/posts.js [hunk 4]
```diff
@@ -1,25 +1,113 @@
...
+          });
+        } catch (e) {
+          console.log('omar',  e);
+        }
+      }
+      case 'submitPostComment': {
+        const post = posts.find((p) => p.post_id === postId);
+        post.comments.push({
+          comment_id: v4(),
+          user_name: 'Marwan Al Shaker',
```
Comment (alshakero) [inline pages/api/posts.js:76]: Usernames can't have spaces. You should use `LOGGED_IN_USER` from [here](https://github.com/alshakerM/my-instagram/blob/add/persistent-updates/stores/constants.js#L1).

File: pages/api/posts.js [hunk 5]
```diff
@@ -1,25 +1,113 @@
...
+          comment: text,
+          comment_likes: 0,
+          is_liked_by_user: false,
+          user_thumbnail:
+            'https://robohash.org/velitillosimilique.png?size=64x64&set=set1',
```
Comment (alshakero) [inline pages/api/posts.js:81]: User comment should be your thumbnail, not a random one.

**Review by alshakero — CHANGES_REQUESTED** [review]
Looks great. One important bug.

File: pages/api/posts.js (hunk 5 above)
Comment (alshakerM) [reply pages/api/posts.js]: Moved index condition.
Changed thumbnail for my pic.
Removed unsafe code.

**Review by alshakero — APPROVED** [review]
Your PRs are really getting nicer and nicer. Great work.

**Comment by alshakerM** [comment]
Moved index condition.
Changed thumbnail for my pic.
Removed unsafe code.
